### PR TITLE
Multi-tenant Linseed changes

### PR DESCRIPTION
**Pull request by Josh-Tigera**
## Description

-Add permissions for Linseed in the management cluster to impersonate Linseed in the managed cluster
-Add environment variables to Linseed deployment so that Linseed will know it is operating in multi-tenant mode

Related CE commit: https://github.com/tigera/calico-private/pull/6553

## For PR author

- [x] Tests for change.
~~- [ ] If changing pkg/apis/, run `make gen-files`~~
~~- [ ] If changing versions, run `make gen-versions`~~

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/render/logstorage/linseed/linseed.go [hunk 1]
```diff
@@ -97,6 +99,8 @@ type Config struct {
 	// Whether this is a management cluster
 	ManagementCluster bool
 
+	MultiTenant bool
```
Comment (caseydavenport) [inline pkg/render/logstorage/linseed/linseed.go:102]: We don't need an explicit option - the code can just check `if c.cfg.Tenant != nil`

File: pkg/render/logstorage/linseed/linseed.go [hunk 2]
```diff
@@ -254,6 +279,7 @@ func (l *linseed) linseedDeployment() *appsv1.Deployment {
 			ValueFrom: secret.GetEnvVarSource(render.ElasticsearchLinseedUserSecret, "password", false),
 		},
 		{Name: "ELASTIC_CA", Value: l.cfg.TrustedBundle.MountPath()},
+		{Name: "MULTI_TENANT", Value: fmt.Sprintf("%v", l.cfg.MultiTenant)},
```
Comment (caseydavenport) [inline pkg/render/logstorage/linseed/linseed.go:282]: Linseed should just check the EXPECTED_TENANT_ID option to determine if it's running in a multi-tenant cluster.

File: pkg/render/logstorage/linseed/linseed_test.go [hunk 3]
```diff
@@ -77,6 +77,7 @@ var _ = Describe("Linseed rendering tests", func() {
 		}
 
 		BeforeEach(func() {
+			replicas = 2
```
Comment (caseydavenport) [inline pkg/render/logstorage/linseed/linseed_test.go:80]: Is this change needed? I'm concerned that it might have been nil before for a reason (but not obvious why)

File: pkg/render/logstorage/linseed/linseed_test.go [hunk 4]
```diff
@@ -159,7 +160,7 @@ var _ = Describe("Linseed rendering tests", func() {
 		})
 
 		It("should render PodAffinity when ControlPlaneReplicas is greater than 1", func() {
-			var replicas int32 = 2
+			replicas = 2
```
Comment (caseydavenport) [inline pkg/render/logstorage/linseed/linseed_test.go:163]: This is a bit confusing - we default replicas to 2, and then also set it to 2 here. Do we need this line at all?

File: pkg/render/logstorage/linseed/linseed_test.go [hunk 5]
```diff
@@ -254,6 +255,45 @@ var _ = Describe("Linseed rendering tests", func() {
 			Expect(d.Spec.Template.Spec.Containers[0].Env).To(ContainElement(corev1.EnvVar{Name: "LINSEED_FIPS_MODE_ENABLED", Value: "true"}))
 		})
 	})
+	Context("multi-tenant rendering", func() {
```
Comment (caseydavenport) [inline pkg/render/logstorage/linseed/linseed_test.go:258]: nit: add a newline before starting a new Context here

File: pkg/render/logstorage/linseed/linseed_test.go [hunk 6]
```diff
@@ -254,6 +255,45 @@ var _ = Describe("Linseed rendering tests", func() {
...
+		var installation *operatorv1.InstallationSpec
+		var replicas int32
+		var cfg *Config
+		clusterDomain := "cluster.local"
+		esClusterConfig := relasticsearch.NewClusterConfig("", 1, 1, 1)
+		BeforeEach(func() {
```
Comment (caseydavenport) [inline pkg/render/logstorage/linseed/linseed_test.go:264]: Same nit here - a newline before `BeforeEach` and `It` clauses as well, just to keep the code a smidge more readable.

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

**Review by caseydavenport — APPROVED** [review]
